Framework/Geometry/inc/MantidGeometry/Instrument/DetectorInfoItem.h

```diff
@@ -0,0 +1,117 @@
+#ifndef MANTID_GEOMETRY_DETECTORINFOITEM_H_
+#define MANTID_GEOMETRY_DETECTORINFOITEM_H_
+
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidKernel/Quat.h"
+#include "MantidKernel/V3D.h"
+
+using Mantid::Geometry::DetectorInfo;
+using Mantid::Kernel::Quat;
+using Mantid::Kernel::V3D;
+
+namespace Mantid {
+namespace Geometry {
+
+/** DetectorInfoItem
+
+DetectorInfoItem is only created by DetectorInfoIterator and allows users of
+the DetectorInfoIterator object access to data from DetectorInfo. The available
+methods include:
+  - isMonitor()
+  - isMaksed()
+  - twoTheta()
+  - position()
+  - rotation()
+
+@author Bhuvan Bezawada, STFC
+@date 2018
+
+Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+
+class MANTID_GEOMETRY_DLL DetectorInfoItem {
+
+public:
+  // Provide copy and move constructors
+  DetectorInfoItem(const DetectorInfoItem &other) = default;
+  DetectorInfoItem &operator=(const DetectorInfoItem &rhs) = default;
+  DetectorInfoItem(DetectorInfoItem &&other) = default;
+  DetectorInfoItem &operator=(DetectorInfoItem &&rhs) = default;
```

Comment:
There is no reason to declare copy/move as `default` in the `public` section --- the would be generate by the compiler anyway. However, we need to prevent copying/moving of `DetectorInfoItem`: It is a helper class that does not contain data and should therefore never live outside the corresponding iterator. Therefore: Declare the copy as `default` in the *private* section, where it is only accessible to the friend.

Reply:
Oh I see - I'll change this just now.

Reply:
Instead of making private. with c++11 and onwards mark these signature as deleted. i.e.
`DetectorInfoItem(DetectorInfoItem &&other) = delete;`

Reply:
I don't think that would work as one of the other classes needs the copy/move constructors. `DetectorInfoPythonIterator` needs access to them.